### PR TITLE
feat: LT-1253 - Add a new contact to a case

### PR DESCRIPTION
**Pull request by jfaracik**
**Story changes**
* Added page and form for viewing and adding additional contacts on a case
* E2Es

**Content and design extras**
* Fixed "Pending" tags being full width
* Combined `team.html` and `own-team.html` - no more duplicated HTML
* Updated flag styling to match new design mockups
* Changed over _some_ `show_pane` components over to the new `Option.components` format
* More content in the content library
* Added angry banner if an HTML page doesn't have a title

### CODE REVIEW

File: cases/forms/respond_to_good_query.py [hunk 1]
```diff
@@ -41,16 +41,20 @@ def respond_to_clc_query_form(request, case):
...
+                                name="control_code",
+                                inset_text=False,
+                            ),
+                        ],
+                    ),
                     Option(key="no", value=cases.RespondClCQueryForm.Controlled.NO),
                 ],
-                classes=["govuk-radios--inline"],
-            ),
-            control_list_entry_question(
```
Comment (BenAndrew-Kainos) [inline cases/forms/respond_to_good_query.py]: why has the clc question been dropped?

File: cases/views/main.py [hunk 2]
```diff
@@ -258,6 +261,32 @@ def get_success_url(self):
         return reverse_lazy("cases:case", kwargs={"pk": self.object_pk})
 
 
+class AdditionalContacts(TemplateView):
+    def get(self, request, **kwargs):
+        """
+        List all documents belonging to a case
+        """
+        case_id = str(kwargs["pk"])
+        case = get_case(request, case_id)
```
Comment (BenAndrew-Kainos) [inline cases/views/main.py]: As you don't seem to need case in additional-contacts.html could you drop this line and instead just pass the case_id in context

File: cases/views/main.py [hunk 3]
```diff
@@ -258,6 +261,32 @@ def get_success_url(self):
...
+            "case": case,
+            "additional_contacts": additional_contacts,
+        }
+        return render(request, "case/views/additional-contacts.html", context)
+
+
+class AddAnAdditionalContact(SingleFormView):
+    def init(self, request, **kwargs):
+        self.object_pk = kwargs["pk"]
+        case = get_case(request, self.object_pk)
```
Comment (BenAndrew-Kainos) [inline cases/views/main.py]: You also don't need case here. Please just pass case id to reduce calls to the API

File: cases/forms/additional_contacts.py [hunk 4]
```diff
@@ -0,0 +1,37 @@
+from django.urls import reverse_lazy
+
+from core.services import get_countries
+from lite_content.lite_internal_frontend.cases import AddAdditionalContact
+from lite_forms.common import country_question
+from lite_forms.components import Form, TextInput, TextArea, BackLink
+
+
+def add_additional_contact_form(request, case):
```
Comment (BenAndrew-Kainos) [inline cases/forms/additional_contacts.py]: This should just take case_id instead of case

File: core/builtins/custom_tags.py [hunk 5]
```diff
@@ -348,3 +349,26 @@ def get_sla_ring_colour(remaining_days):
 def is_exhibition(case_type):
     result = True if case_type == CaseType.EXHIBITION else False
     return result
+
+
+@register.simple_tag
+@mark_safe
+def missing_title():
```
Comment (BenAndrew-Kainos) [inline core/builtins/custom_tags.py]: Nice idea. Hopefully people will notice the fact they've missed a title now

File: templates/case/advice/user.html [hunk 6]
```diff
@@ -6,31 +6,35 @@
 	{% include "case/views/includes/breadcrumbs.html" with text="cases.ApplicationPage.Actions.ADVICE" %}
 {% endblock %}
 
+{% block title %}
+	{% lcs 'cases.ViewAdvicePage.USER_ADVICE' %} -
+	{% lcs 'cases.ApplicationPage.Actions.ADVICE' %}
+{% endblock %}
```
Comment (BenAndrew-Kainos) [inline templates/case/advice/user.html]: Could we group final.html, team.html & user.html better. They seem to duplicate html a lot (maybe making a advice.html page they extend is a good move)

File: templates/case/give-advice.html [hunk 7]
```diff
@@ -81,7 +81,7 @@ <h2 class="govuk-error-summary__title" id="error-summary-title">
 								{% svg 'new' %}
 								{% lcs 'cases.Advice.IMPORT_PROVISO' %}
 							</a>
-						</div
+						</div>
```
Comment (BenAndrew-Kainos) [inline templates/case/give-advice.html]: Good spot

File: ui_automation_tests/features/additional_contacts.feature [hunk 8]
```diff
@@ -0,0 +1,13 @@
+@all @internal @additional_contacts
+Feature: I want to add additional contacts to a case
+    So that all information is available about a case
+
+  @LT_1253_Additional_contacts @regression
+  Scenario: Add an additional contact to a case
+    Given I sign in to SSO or am signed into SSO
+    And I create standard application or standard application has been previously created
```
Comment (phlptt) [inline ui_automation_tests/features/additional_contacts.feature]: Better to use open licence for this as they dont require any document uploads

File: templates/case/views/additional-contacts.html [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+							{{ contact.phone_number }}
+						</td>
+					</tr>
+				{% endfor %}
+			</tbody>
+		</table>
+	{% else %}
+		{% include "includes/notice.html" with text='cases.AdditionalContacts.NO_CONTENT_NOTICE' %}
+	{% endif %}
+{% endblock %}
```
Comment (BenAndrew-Kainos) [inline templates/case/views/additional-contacts.html]: newline

File: templates/teams/team.html [hunk 10]
```diff
@@ -1,52 +1,66 @@
 {% extends 'layouts/base.html' %}
 
-{% load humanize %}
-
 {% block back_link %}
-	<a href="/" class="govuk-back-link">Back</a>
+	{% if not is_user_in_team %}
+		<a href="/" class="govuk-back-link">Back</a>
```
Comment (BenAndrew-Kainos) [inline templates/teams/team.html]: Specific link please & move text to content

File: ui_automation_tests/pages/application_page.py [hunk 11]
```diff
@@ -102,6 +103,11 @@ def click_case_officer_button(self):
         scroll_to_element_by_id(self.driver, self.CASE_OFFICER_ID)
         self.driver.find_element_by_id(self.CASE_OFFICER_ID).click()
 
+    def click_additional_contacts_link(self):
+        self.click_drop_down()
```
Comment (BenAndrew-Kainos) [inline ui_automation_tests/pages/application_page.py]: Could we specify the dropdown with an ID too please

File: ui_automation_tests/step_defs/test_additional_contacts.py [hunk 12]
```diff
@@ -0,0 +1,42 @@
...
+    additional_contacts_page.enter_country("United Kingdom")
+    additional_contacts_page.enter_email(context.additional_contact_email)
+    additional_contacts_page.enter_phone_number(fake.phone_number())
+
+    functions.click_submit(driver)
+
+
+@then("I can see the new contact in the list")
+def click_post_note(driver, context):
+    assert context.additional_contact_email in Shared(driver).get_text_of_body()
```
Comment (BenAndrew-Kainos) [inline ui_automation_tests/step_defs/test_additional_contacts.py]: Never ever use text of body. Please specify the area

File: ui_automation_tests/step_defs/test_eua_queries.py [hunk 13]
```diff
@@ -25,9 +25,10 @@ def dropdown_contains_correct_functionality(driver):
     assert application_page.get_case_officer_element()
     assert application_page.get_generate_document_element()
     assert application_page.get_assign_user_element()
+    assert application_page.get_additional_contacts_element()
     # This tests that the expected elements are the only ones that appear, and that any new functionality added
     # should be tested if not tested elsewhere.
-    assert application_page.get_action_dropdown_entries_count() == 7
+    assert application_page.get_action_dropdown_entries_count() == 8
```
Comment (BenAndrew-Kainos) [inline ui_automation_tests/step_defs/test_eua_queries.py]: Where does this 8 come from?

File: ui_automation_tests/features/additional_contacts.feature (hunk 8 above)
Comment (jfaracik) [reply ui_automation_tests/features/additional_contacts.feature]: Done!

File: ui_automation_tests/step_defs/test_eua_queries.py (hunk 13 above)
Comment (jfaracik) [reply ui_automation_tests/step_defs/test_eua_queries.py]: Eight is the number of items in the Actions menu. It won't be here much longer due to the case UI redesign.

File: cases/views/main.py (hunk 2 above)
Comment (jfaracik) [reply cases/views/main.py]: It's used for the breadcrumb bar

File: cases/forms/respond_to_good_query.py (hunk 1 above)
Comment (jfaracik) [reply cases/forms/respond_to_good_query.py]: It hasn't, it's just been moved up inside the Option. `show_pane` is deprecated.

File: ui_automation_tests/pages/application_page.py (hunk 11 above)
Comment (jfaracik) [reply ui_automation_tests/pages/application_page.py]: The dropdown is going to be removed in the upcoming case UI redesign.

File: cases/views/main.py (hunk 2 above)
Comment (BenAndrew-Kainos) [reply cases/views/main.py]: Damn that seems inefficient. Do we really need the reference name at the expense of fetching the entire case. At the very least could we use a tiny serializer

File: cases/views/main.py (hunk 3 above)
Comment (BenAndrew-Kainos) [reply cases/views/main.py]: This has been resolved without action

File: cases/forms/additional_contacts.py (hunk 4 above)
Comment (BenAndrew-Kainos) [reply cases/forms/additional_contacts.py]: This has been resolved without action

File: ui_automation_tests/pages/application_page.py (hunk 11 above)
Comment (BenAndrew-Kainos) [reply ui_automation_tests/pages/application_page.py]: Please leave a comment

File: ui_automation_tests/step_defs/test_eua_queries.py (hunk 13 above)
Comment (BenAndrew-Kainos) [reply ui_automation_tests/step_defs/test_eua_queries.py]: Please remove this line.

**Comment by jfaracik** [comment]
> There are also 5 comments from the previous review that have no action

Aha this'll be why:

image


GitHub didn't push my changes

File: ui_automation_tests/pages/application_page.py (hunk 11 above)
Comment (jfaracik) [reply ui_automation_tests/pages/application_page.py]: I've added a comment to the function but I don't see it providing value.

File: ui_automation_tests/step_defs/test_eua_queries.py (hunk 13 above)
Comment (jfaracik) [reply ui_automation_tests/step_defs/test_eua_queries.py]: Again done it, but I don't see it providing value.

File: cases/views/main.py (hunk 2 above)
Comment (jfaracik) [reply cases/views/main.py]: It's certainly an idea, one that we can toy with in the case redesign story.

File: templates/case/advice/user.html (hunk 6 above)
Comment (jfaracik) [reply templates/case/advice/user.html]: I agree! I'll do it as part of the case redesign.

File: cases/views/main.py (hunk 2 above)
Comment (BenAndrew-Kainos) [reply cases/views/main.py]: Obviously we do this in a lot of places so we should consider API load when redesigning all pages

File: templates/case/advice/user.html (hunk 6 above)
Comment (BenAndrew-Kainos) [reply templates/case/advice/user.html]: Cool

File: ui_automation_tests/pages/application_page.py (hunk 11 above)
Comment (BenAndrew-Kainos) [reply ui_automation_tests/pages/application_page.py]: Sorry I meant a TODO to include an ID when the redesign happens

**Pull request merged**